plugin/errors.go

```diff
@@ -41,9 +42,83 @@ type InstallEqualVersionError struct {
 	Msg string
 }
 
+// Error returns the error message.
 func (e InstallEqualVersionError) Error() string {
 	if e.Msg != "" {
 		return e.Msg
 	}
 	return "installing plugin with version equal to the existing plugin version"
 }
+
+// PluginUnknownError is used when there is an issue executing a plugin
+// and should be reported to Notation developers.
+type PluginUnknownError struct {
+	Msg        string
+	InnerError error
+}
+
+// Error returns the error message.
+func (e PluginUnknownError) Error() string {
+	if e.Msg != "" {
+		return e.Msg
+	}
+	return e.InnerError.Error()
+}
+
+// Unwrap returns the inner error.
+func (e PluginUnknownError) Unwrap() error {
+	return e.InnerError
+}
+
+// PluginMalformedError is used when there is an issue with plugin and
+// should be fixed by plugin developers.
+type PluginMalformedError struct {
+	Msg        string
+	InnerError error
+}
+
+// Error returns the error message.
+func (e PluginMalformedError) Error() string {
+	if e.Msg != "" {
+		return e.Msg
+	}
+	return e.InnerError.Error()
+}
+
+// Unwrap returns the inner error.
+func (e PluginMalformedError) Unwrap() error {
+	return e.InnerError
+}
+
+// PluginDirectryWalkError is used when there is an issue with plugins directory
+// and should suggest user to check the permission of plugin directory.
+type PluginDirectryWalkError struct {
+	Err error
+}
+
+// Error returns the error message.
+func (e PluginDirectryWalkError) Error() string {
+	return e.Err.Error()
+}
+
+// Unwrap returns the inner error.
+func (e PluginDirectryWalkError) Unwrap() error {
+	return e.Err
+}
+
+// PluginExectableFileError is used when there is an issue with plugin
+// executable file and should suggest user to check the existence, permission
+// and platform/arch compatibility of plugin.
+type PluginExectableFileError struct {
+	Err error
```

Comment:
same here. can we use a `var` for this error type instead?

Reply:
updated.